purchases/src/test/java/com/revenuecat/purchases/paywalls/events/PaywallEventSerializationTests.kt

```diff
@@ -0,0 +1,67 @@
+package com.revenuecat.purchases.paywalls.events
+
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import com.revenuecat.purchases.ExperimentalPreviewRevenueCatPurchasesAPI
+import kotlinx.serialization.decodeFromString
+import kotlinx.serialization.encodeToString
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.Test
+import org.junit.runner.RunWith
+import java.util.Date
+import java.util.UUID
+
+@OptIn(ExperimentalPreviewRevenueCatPurchasesAPI::class)
+@RunWith(AndroidJUnit4::class)
+class PaywallEventSerializationTests {
+
+    private val event = PaywallStoredEvent(
+        event = PaywallEvent(
+            creationData = PaywallEvent.CreationData(
+                id = UUID.fromString("298207f4-87af-4b57-a581-eb27bcc6e009"),
+                date = Date(1699270688884)
+            ),
+            data = PaywallEvent.Data(
+                offeringIdentifier = "offeringID",
+                paywallRevision = 5,
+                sessionIdentifier = UUID.fromString("315107f4-98bf-4b68-a582-eb27bcb6e111"),
+                displayMode = "footer",
+                localeIdentifier = "es_ES",
+                darkMode = true
+            ),
+            type = PaywallEventType.IMPRESSION,
+        ),
+        userID = "testAppUserId",
+    )
+
+    @Test
+    fun `can encode paywall event correctly`() {
+        val eventString = PaywallEventsManager.json.encodeToString(event)
+        assertThat(eventString).isEqualTo(
```

Comment:
I imagine you compared these with iOS 👍🏻

Reply:
These are different than in iOS, but it's the way they are stored in disk, so it shouldn't matter if they are different I think. In iOS, the encoding is pretty different, but all the data is there.

Reply:
Oh yeah of course never mind